mech_client/interact.py

```diff
@@ -281,4 +302,27 @@ def interact(
         prompt=prompt,
         tool=tool,
     )
-    return watch_for_events(wss=wss, mech_contract=mech_contract, ledger_api=ledger_api)
+    request_id = watch_for_request_id(
+        wss=wss, mech_contract=mech_contract, ledger_api=ledger_api
+    )
+    print(f"Created on-chain request with ID {request_id}")
+    if agent_address is not None:
+        print(f"Requesting data from agent {agent_address}")
+        loop = asyncio.new_event_loop()
+        task = loop.create_task(
+            request_mech_for_data(
+                request_id=request_id,
+                agent_address=agent_address,
+                crypto=crypto,
+            )
+        )
+        loop.run_until_complete(task)
+        data = task.result()
+    else:
+        data = watch_for_data(
+            request_id=request_id,
+            wss=wss,
+            mech_contract=mech_contract,
+            ledger_api=ledger_api,
+        )
```

Comment:
If the agent address is provided we request data directly from agent using ACN, else we use websocket subscription